### ISSUE TITLE
Possibilty to port into javascript/java/c++

### ISSUE THREAD

**Issue by michaellee8** [RobotGame/rgserver#44]
It is sort of great fun with this game, but the language seems to be too narrow in terms of supported languages. I think it can we great if there can be any support to other languages, like the most popular javascript/java/c++ nowadays. Or even a crazy thought of expose the API for offline computation?

**Comment by mueslo** [comment]
I have a bit of local work on this, i.e. a de/serializer for robot actions and game states and match executer. I wrote a simple network script where 2 bots can play against each other on a central server. I hope I can clean up the code and push it to Github in the next few weeks.

**Comment by WhiteHalmos** [comment]
I also had some ideas about how to implement this. We can effectively change to a file-based system where each bot takes as input the game state and outputs moves for all its bots (also switching to calling per-turn rather than per-bot). But alas, no time to implement, so if anyone (looking at @mueslo) actually implements it I'm happy to switch to it. Then we can have almost any language, and it's just sandboxing that's hard.

**Comment by michaellee8** [comment]
May be we can do further, we may just let our AI controller-sided, that is, we only have one instance of program to control all the bots, it makes life easier since some tasks of the robot are repeated in every robot move. May be we can have some new implementation that our AI system become an interactive program which communicates by stdio and stdout queries.

**Comment by michaellee8** [comment]
If we can have our robots' moving plans stored in our program already then it will be more efficient in terms of reaources.

P. S.: I am really interested on developing this gaming system, I will consider redeveloping the core of the system when I have finished the 2 projects that are on my hands currently, maybe we need some sort of online judging system if we want to do the compile and run way for the robots.

**Comment by WhiteHalmos** [comment]
Yes, by per-turn I mean switch to calling "actTurn" rather than "act" as it is right now, and return a dictionary of actions for each bot location.
